weather-app/app/components/Sidebar.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+import React, { useState } from "react";
+import styles from "./Sidebar.module.scss";
+
+const Sidebar = () => {
+  const [isWeatherActive, setIsWeatherActive] = useState<boolean>(true);
+  const [isFavoriteListActive, setIsFavoriteListActive] =
+    useState<boolean>(false);
+
+  const weatherIconHandler = () => {
+    if (!isWeatherActive) {
+      setIsWeatherActive(true);
+      setIsFavoriteListActive(false);
+    }
+  };
+
+  const favoriteListIconHandler = () => {
+    if (!isFavoriteListActive) {
+      setIsWeatherActive(false);
+      setIsFavoriteListActive(true);
+    }
+  };
+
+  return (
+    <div className={styles.sidebar}>
+      <Link className={styles.link} href={"/"} onClick={weatherIconHandler}>
+        <Image
+          className={styles.img}
+          src={isWeatherActive ? "/weather-icon.svg" : "/weather-icon-pale.svg"}
```

Comment:
Now that we have a `pathname`, we can use it as a condition instead of `isWeatherActive`.
`src={pathname === "/" ? "/weather-icon.svg" : "/weather-icon-pale.svg"}`

I really like your way to use ternary and the icon here! I can take the same approach for the star icon.

Reply:
Thanks!
I modified it.